internal/config/net_test.go

```diff
@@ -253,77 +373,190 @@ func TestNet_Bind(t *testing.T) {
 			if test.checkFunc == nil {
 				test.checkFunc = defaultCheckFunc
 			}
-			t := &Net{
-				DNS:    test.fields.DNS,
-				Dialer: test.fields.Dialer,
-				TLS:    test.fields.TLS,
+			s := &SocketOption{
+				ReusePort:                test.fields.ReusePort,
+				ReuseAddr:                test.fields.ReuseAddr,
+				TCPFastOpen:              test.fields.TCPFastOpen,
+				TCPNoDelay:               test.fields.TCPNoDelay,
+				TCPCork:                  test.fields.TCPCork,
+				TCPQuickAck:              test.fields.TCPQuickAck,
+				TCPDeferAccept:           test.fields.TCPDeferAccept,
+				IPTransparent:            test.fields.IPTransparent,
+				IPRecoverDestinationAddr: test.fields.IPRecoverDestinationAddr,
 			}
 
-			got := t.Bind()
+			got := s.Bind()
 			if err := test.checkFunc(test.want, got); err != nil {
 				tt.Errorf("error = %v", err)
 			}
 		})
 	}
 }
 
-func TestNet_Opts(t *testing.T) {
+func TestSocketOption_ToSocketFlag(t *testing.T) {
+	t.Parallel()
 	type fields struct {
-		DNS    *DNS
-		Dialer *Dialer
-		TLS    *TLS
+		socketOpts *SocketOption
 	}
 	type want struct {
-		want []net.DialerOption
+		want control.SocketFlag
 	}
 	type test struct {
 		name       string
 		fields     fields
 		want       want
-		checkFunc  func(want, []net.DialerOption) error
+		checkFunc  func(want, control.SocketFlag) error
 		beforeFunc func()
 		afterFunc  func()
 	}
-	defaultCheckFunc := func(w want, got []net.DialerOption) error {
+	defaultCheckFunc := func(w want, got control.SocketFlag) error {
 		if !reflect.DeepEqual(got, w.want) {
 			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
 		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       fields: fields {
-		           DNS: DNS{},
-		           Dialer: Dialer{},
-		           TLS: TLS{},
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           fields: fields {
-		           DNS: DNS{},
-		           Dialer: Dialer{},
-		           TLS: TLS{},
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name:   "return flg when socketOpts is nil",
+			fields: fields{},
+			want: want{
+				want: 0x00,
+			},
+		},
+		{
+			name: "return flg when socketOpts.ReuseAddr true and others are false",
+			fields: fields{
+				socketOpts: &SocketOption{
+					ReuseAddr: true,
+				},
+			},
+			want: want{
+				want: 0x2,
```

Comment:
the flag is os dependent variable, please use the flag defined in the control package.
Please also fix another test cases

Reply:
thx for the review. could you please more detail about what you want to say?
I used the `uint` value for the checking output value.
and here is the definition.
https://github.com/vdaas/vald/blob/4bf65f85dcaa6c28f7b755b2a1410322f86ccbbf/internal/net/control/control.go#L46-L56
I'm not sure `the flag is os dependent variable`.

Reply:
sorry I misread the implementation. there are no problem in the current test implementation.